common_types/src/datum.rs

```diff
@@ -176,6 +176,28 @@ impl DatumKind {
     pub fn into_u8(self) -> u8 {
         self as u8
     }
+
+    /// Return None for variable-length type
+    pub fn size(&self) -> Option<usize> {
+        let size = match self {
+            DatumKind::Null => 1,
```

Comment:
https://arrow.apache.org/docs/format/Columnar.html#null-layout

Not sure but according to the spec NULL type's size is 0?